src/connection.rs

```
@@ -1,23 +1,18 @@
-use crate::{cluster::SignerRaftNode, error::SignerError};
+use crate::error::SignerError;
 use ed25519_consensus::SigningKey;
-use log::{error, info, warn};
+use log::{error, info};
+use std::net::TcpStream;
 use std::thread::sleep;
 use std::time::Duration;
-use std::{net::TcpStream, sync::Arc};
 use tendermint_p2p::secret_connection::{self, SecretConnection};
 
 pub fn open_secret_connection(
     host: &str,
     port: u16,
     identity_key: SigningKey,
     protocol_version: secret_connection::Version,
-    raft_node: &Arc<SignerRaftNode>,
 ) -> Result<SecretConnection<TcpStream>, SignerError> {
     loop {
-        if !raft_node.is_leader() {
-            warn!("Not the leader");
-            return Err(SignerError::TODO);
-        }
```

so this is one of the places where it might have been beneficial to fail fast if we are not a leader, but it is prone to toctou errors, im okay with removing it